typings/index.d.ts

```diff
@@ -521,9 +521,9 @@ export abstract class Channel extends Base {
   public id: Snowflake;
   public readonly partial: false;
   public type: keyof typeof ChannelTypes;
-  public delete(): Promise<Channel>;
-  public fetch(force?: boolean): Promise<Channel>;
-  public isText(): this is TextBasedChannels;
+  public delete(): Promise<this>;
+  public fetch(force?: boolean): Promise<this>;
```

Comment:
Didn’t @suneettipirneni say something about returning `this` and how it didn’t make sense in this situation?

Reply:
It makes sense here, because if you narrow a type the promise on this class has the narrowed return type as well.

Reply:
Oh alright, when does it not make sense then?

Reply:
The other PR was more of an aesthetic thing, the type that was discussed before wasn't narrowed. But even in the case before it's fine.

Reply:
Alright, thanks